src/Dotnet.Script.DependencyModel/ScriptPackage/ScriptFilesDependencyResolver.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System;
+using Dotnet.Script.DependencyModel.Logging;
+
+namespace Dotnet.Script.DependencyModel.ScriptPackage
+{
+    public class ScriptFilesDependencyResolver
+    {
+        private readonly Logger _logger;
+
+        private static readonly Regex TargetFrameworkMatcher =
+            new Regex(@"^.*(?:contentFiles|content)[\/,\\]csx[\/,\\](.*)[\/,\\].*$",
+                RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        private static readonly Regex RootPathMatcher =
+            new Regex(@"(^.*(?:contentFiles|content)[\/,\\]csx[\/,\\].*[\/,\\]).*$",
+                RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        public ScriptFilesDependencyResolver(LogFactory logFactory)
+        {            
+            _logger = logFactory.CreateLogger<ScriptFilesDependencyResolver>();
+        }
+
+        public string[] GetScriptFileDependencies(string path, string[] nugetPackageFolders)
+        {
+            var result = new List<string>();
+            var packagePath = GetPackageFullPath(path, nugetPackageFolders);
+            var allScriptFiles = Directory.GetFiles(packagePath, "*.csx", SearchOption.AllDirectories);
```

Comment:
should we perhaps discover script files only once, somewhere in the beginning and pass that list to both `ScriptFilesDependencyResolver` and `ScriptProjectProvider`?

Reply:
oh wait, this happens per nuget folder, ok